src/libraries/System.IO.Ports/src/System/IO/Ports/SerialStream.Unix.cs

```diff
@@ -715,7 +729,7 @@ private void RaiseDataReceivedEof()
 
         private unsafe int ProcessRead(SerialStreamIORequest r)
         {
-            Span<byte> buff = r.Buffer.Span;
+            ReadOnlySpan<byte> buff = r.Buffer.Span;
```

Comment:
The only reason this is working is because we pass this to native code... This should remain `Span`, same for `Memory` change below.

Reply:
Spoke with @krwq offline and I decided to revert the changes in this file as a suitable implementation for the overloads without compromising the existing functionality requires more changes than I would want to make as part of this PR. Will file an issue to add the overloads.